### PR TITLE
[PHPStan] Update PHPStan to 0.12.95

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
It seems cause error in  test:

```diff
1) Rector\Tests\TypeDeclaration\Rector\ClassMethod\AddArrayReturnDocTypeRector\AddArrayReturnDocTypeRectorTest::test with data set #19 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
66
rules-tests/TypeDeclaration/Rector/ClassMethod/AddArrayReturnDocTypeRector/Fixture/configuration_with_string_string_array.php.inc
67
Failed asserting that string matches format description.
68
--- Expected
69
+++ Actual
70
@@ @@
71
 final class ConfigurationWithStringStringArray
72
 {
73
     /**
74
-     * @return array<class-string<\Rector\Transform\Rector\Assign\DimFetchAssignToMethodCallRector>, array<string, \Rector\Transform\ValueObject\DimFetchAssignToMethodCall[]>>
75
+     * @return array<class-string<\Rector\Transform\Rector\Assign\DimFetchAssignToMethodCallRector>, array<string, \Rector\Transform\ValueObject\DimFetchAssignToMethodCall[]>&mixed[]>
76
      */
```

**Comment by samsonasik** [comment]
Fixed 🎉

File: src/Application/VersionResolver.php [hunk 1]
```diff
@@ -38,7 +39,10 @@ public static function resolvePackageVersion(): string
         return trim($version, '"');
     }
 
-    public static function resolverReleaseDateTime(): DateTime
+    /**
+     * @return DateTime|DateTimeImmutable
+     */
+    public static function resolverReleaseDateTime(): \DateTime
```
Comment (samsonasik) [inline src/Application/VersionResolver.php]: @Bl00D4NGEL it seems needs to be skipped as the method actually returns the exact DateTime object.

Comment (samsonasik) [reply src/Application/VersionResolver.php]: resolved at https://github.com/rectorphp/rector-src/pull/730

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: src/Application/VersionResolver.php (hunk 1 above)
Comment (Bl00D4NGEL) [reply src/Application/VersionResolver.php]: I'm not sure if this is actually "expected"
Given your fixture skip_exactly_return_datetime.php.inc
I would expect that if a function returns "just a DateTime" I would still expect the function signature (specifically the return type) to change. Maybe I'm just dumb and don't understand why this change was necessary, can you maybe explain this to be a little more so I can understand, too?

Comment (samsonasik) [reply src/Application/VersionResolver.php]: the docblock make phpstan error

**Pull request merged**